operator_csv_libs/csv.py

```diff
@@ -108,18 +108,24 @@ def add_image_pullsecret(self, name):
         """ Add image pull secret for all operator deployment deployments. Existing pull secrets will be kept
 
         :param name: String or list of strings containing name of the pull secret to add
-        :type name: string, list 
+        :type name: string, list
         """
         if isinstance(name, str):
             p = [{'name': name}]
         else:
             p = [ {'name': x} for x in name ]
-        
+
         for d in self.csv['spec']['install']['spec']['deployments']:
             if not 'imagePullSecrets' in d['spec']['template']['spec']:
-                d['spec']['template']['spec']['imagePullSecrets'] = []
-            # Add the new pull secrets
-            d['spec']['template']['spec']['imagePullSecrets'] = p    
+                # If imagepullsecret is missing, set it
+                d['spec']['template']['spec']['imagePullSecrets'] = p
+            else:
+                # If imagepullsecret exists, add to the list
+                if not(all(x in p for x in d['spec']['template']['spec']['imagePullSecrets'])):
+                    if isinstance(d['spec']['template']['spec']['imagePullSecrets'], list):
+                        d['spec']['template']['spec']['imagePullSecrets'].extend(p)
+                    else:
+                        raise Exception('imagePullSecrets is not of type list')   
```

Comment:
Might be better to add the `TypeError` exception, then a generic `Exception`?

Reply:
Will change that!